app/main.py

```diff
@@ -1,8 +1,67 @@
+from typing import List
+
+
 class Car:
-    # write your code here
-    pass
+    def __init__(self,
+                 comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(
+            self,
+            distance_from_city_center: float,
+            clean_power: int,
+            average_rating: float,
+            count_of_ratings: int
+    ) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: List) -> float:
+        total_income = 0.0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income = self.wash_single_car(car)
+                total_income += income
+        return round(total_income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        if not isinstance(car, Car):
+            raise ValueError("The argument must be a Car instance")
+        if car.clean_mark >= self.clean_power:
+            return 0.0
+        washing_price = (
+            car.comfort_class
+            * (self.clean_power - car.clean_mark)
+            * self.average_rating
+            / self.distance_from_city_center
+        )
+        return round(washing_price, 1)
+
+    def wash_single_car(self, car: Car) -> float:
+        if not isinstance(car, Car):
+            raise ValueError("The argument must be a Car instance")
+
+        if car.clean_mark < self.clean_power:
+            price = self.calculate_washing_price(car)
+            car.clean_mark = self.clean_power
+            return price
+        return 0.0
+
+    def rate_service(self, new_rating: float) -> None:
+        if not (1.0 <= new_rating <= 5.0):
+            raise ValueError("new_rating must be between 1.0 and 5.0")
+
+        total_ratings = self.average_rating * self.count_of_ratings
+        new_total_ratings = total_ratings + new_rating
+        self.count_of_ratings += 1
+        self.average_rating = round(
```

Comment:
you can calculate new average rating in one statement without creating extra variables